sources/packages/backend/apps/api/src/route-controllers/education-program-offering/_tests_/education-program-offering.institutions.createOffering.e2e-spec.ts

```diff
@@ -0,0 +1,163 @@
+    app = nestApplication;
+    db = createE2EDataSources(dataSource);
+    const { institution, user: institutionUser } = await getAuthRelatedEntities(
+      db.dataSource,
+      InstitutionTokenTypes.CollegeFUser,
+    );
+    collegeF = institution;
+    collegeFUser = institutionUser;
+    collegeFLocation = createFakeInstitutionLocation(collegeF);
+    await authorizeUserTokenForLocation(
+      db.dataSource,
+      InstitutionTokenTypes.CollegeFUser,
+      collegeFLocation,
+    );
+  });
+
+  it("Should create a new offering when passed valid data.", async () => {
+    // Arrange
+    const institutionUserToken = await getInstitutionToken(
+      InstitutionTokenTypes.CollegeFUser,
+    );
+
+    const fakeEducationProgram = createFakeEducationProgram(
+      collegeF,
+      collegeFUser,
+    );
+    fakeEducationProgram.sabcCode = "nbgvc";
+    const savedFakeEducationProgram = await db.educationProgram.save(
+      fakeEducationProgram,
+    );
+    const endpoint = `/institutions/education-program-offering/location/${collegeFLocation.id}/education-program/${savedFakeEducationProgram.id}`;
+    const payload = {
+      offeringName: "Offering 1",
+      yearOfStudy: 1,
+      showYearOfStudy: true,
+      offeringIntensity: OfferingIntensity.fullTime,
+      offeringDelivered: OfferingDeliveryOptions.Onsite,
+      hasOfferingWILComponent: "no",
+      studyStartDate: "2023-09-01",
+      studyEndDate: "2024-06-30",
+      lacksStudyBreaks: false,
+      studyBreaks: [
+        {
+          breakStartDate: "2023-12-01",
+          breakEndDate: "2024-01-01",
+          breakDays: 32,
+          eligibleBreakDays: 21,
+          ineligibleBreakDays: 11,
+        },
+      ],
+      studyBreaksOverlappingDetection: false,
+      studyPeriodBreakdown: {
+        totalDays: 304,
+        totalFundedWeeks: 42,
+        fundedStudyPeriodDays: 293,
+        unfundedStudyPeriodDays: 11,
+      },
+      offeringType: OfferingTypes.Public,
+      offeringDeclaration: true,
+      mode: "editable",
+      validationWarnings: [],
+      validationInfos: ["invalidStudyBreakAmountOfDays"],
+      applicationId: "",
+      applicationStatus: "",
+      actualTuitionCosts: 1234,
+      programRelatedCosts: 3211,
+      mandatoryFees: 456,
+      exceptionalExpenses: 555,
+    };
+
+    // Act/Assert
+    let educationProgramOfferingId: number;
+    await request(app.getHttpServer())
+      .post(endpoint)
+      .send(payload)
+      .auth(institutionUserToken, BEARER_AUTH_TYPE)
+      .expect(HttpStatus.CREATED)
+      .then((response) => {
+        expect(response.body.id).toBeGreaterThan(0);
+        educationProgramOfferingId = response.body.id;
+      });
+
+    const createdEducationProgramOffering =
+      await db.educationProgramOffering.findOne({
+        where: { id: educationProgramOfferingId },
+      });
+    expect(createdEducationProgramOffering).toEqual(
+      expect.objectContaining({
```

Comment:
Can we use the regular object comparison here?

Reply:
No, because there are some properties that bring dates created in the backend for example. They not important to test in this case.

Reply:
Got it. Thanks Andre.